aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -330,13 +330,17 @@ blockReduce(AccumT* smem, AccumT val,
   AccumT warpVal = defaultVal;
 
   // First warp will perform per-warp reductions for the remaining warps
+  uint32_t mask = (((uint64_t)1) << (blockDim.x / 32)) - 1;
```

Comment:
Just curious, what's the reason for masking in this case, instead of `__syncwarp()` with the full mask by default?

Reply:
Also I think there's no need for uint64_t as overflow doesn't affect lower bits.

Reply:
The way the code is structured, not all threads will be participating in the __syncwarp, so calling it with the full mask will be illegal. I'd prefer not to shift by more bits than exist in the operand, even though nominally one doesn't care about overflow in this case. Perf implications are negligible.